course_discovery/apps/api/utils.py

```diff
@@ -49,6 +52,33 @@ def get_query_param(request, name):
     return cast2int(request.query_params.get(name), name)
 
 
+def update_query_params_by_body_data(func_to_decorate):
```

Comment:
```suggestion
def update_query_params_with_body_data(func_to_decorate):
```
I think `with` would be a better word to use here for clarity.

Reply:
Let's improve the name. Good.